src/ScopedAssign.h

```diff
@@ -0,0 +1,25 @@
+#ifndef SCOPEDASSIGN_HH
+#define SCOPEDASSIGN_HH
+
+/** Assign new value to some variable and restore the original value
+  * when this object goes out of scope.
+  */
+template<typename T> class ScopedAssign
+{
+public:
+	[[nodiscard]] ScopedAssign(T& var_, T newValue)
```

Comment:
Thanks a lot for your suggestion to use [[nodiscard]] here! I plan to import this improvement in the openMSX code.
But not just yet ...

Strictly according to the standard, [[nodiscard]] on constructors is only available starting from c++20 (not yet in c++17). Starting from gcc-10, it is supported, and even in c++17 mode. Here is an experiment I did: https://godbolt.org/z/sjb69fvaE

In openMSX we plan to move to c++20 soonish (hopefully in a few months, right after the openmsx-18 release). Then I'll import this change.

For me it's fine if you keep this attribute in the debugger code. It means that if you use a too old compiler (gcc-9, clang-9) you'll get a warning on the `ScopedAssign` constructor (and not on incorrect use of this constructor).

Reply:
Funny that not even `-Wall -pedantic` detects that on gcc-11, just lots of extra `;` in Q_OBJECT basically.